Xamarin.Essentials/Types/LocationExtensions.android.cs

```diff
@@ -25,9 +25,11 @@ public static partial class LocationExtensions
             {
                 Latitude = location.Latitude,
                 Longitude = location.Longitude,
-                Altitude = location.HasAltitude ? location.Altitude : (double?)null,
+                Altitude = location.HasAltitude ? location.Altitude : default(double?),
                 TimestampUtc = location.GetTimestamp().ToUniversalTime(),
-                Accuracy = location.HasAccuracy ? location.Accuracy : (float?)null
+                Accuracy = location.HasAccuracy ? location.Accuracy : default(float?),
+                Bearing = location.HasBearing && location.Bearing != 0.0d ? location.Bearing : default(double?),
+                Speed = location.HasSpeed && location.Speed == 0.0d ? location.Speed : default(double?)
```

Comment:
Ternary operations like this scare me, Can you add additional () in there... Why do we need to check against 0.0d?

Reply:
so 0.0 means it's unavailable on android.  We should probably check if it's nearly equal to 0 though instead because precision.

Reply:
Like `Math.Abs(location.Bearing) < 0.000001` or something.

Reply:
The exact null comparison works. It returns positive zero if no data is available. If you insist I will change this to use delta equals

Reply:
@jamesmontemagno  whoops, good catch! changed equality and added braces to clarify operator precedence